PyRoute/Star.py

```diff
@@ -197,50 +223,72 @@ def parse_line_into_star(line, sector, pop_code, ru_calc):
 
     @staticmethod
     def _parse_line_into_star_core(star, line, sector, pop_code, ru_calc):
```

Comment:
I ended up moving this entire function into an equivalent of a ` ParseStarInput` class. It keeps all the parsing bits together, and the `Star` class is more about managing the data during processing. Consider moving this, as it may simplify and isolate the different parts of the code.

Reply:
Good catch.

Reply:
Your comment tripped a little reorganisation.

The old `ParseStarInput` class contained only fallback parsing, so I renamed it to `ParseStarInputFallback` (now it does what it says on the tin).

The core of the primary starline parsing, plus supporting arrays, regex, etc, is now in a new `ParseStarInput` class.